assets/js/components/user-input/UserInputKeywords.js

```diff
@@ -85,30 +86,59 @@ export default function UserInputKeywords( { slug, max } ) {
 		] );
 	}, dependencies );
 
-	const onKeyDown = useCallback( ( index, { keyCode } ) => {
+	const onKeyDown = useCallback( ( index, { keyCode, target } ) => {
 		const nonEmptyValues = values.filter( ( value ) => value.length > 0 );
+		const nonEmptyValuesLength = nonEmptyValues.length;
+
 		if ( ( keyCode === ENTER || keyCode === COMMA ) && nonEmptyValues.length < max ) {
 			updateKeywords( [
 				...values.slice( 0, index + 1 ),
 				'',
 				...values.slice( index + 1 ),
 			] );
 
-			setTimeout( () => {
-				const input = keywordsContainer.current.querySelector( `#${ slug }-keyword-${ index + 1 }` );
-				if ( input ) {
-					input.focus();
-				}
-			}, 50 );
+			setCanDeleteKeyword( true ); // New keyword has been added, so hitting backspace now will remove it
+			setTimeout( () =>
+				focusInput( `#${ slug }-keyword-${ index + 1 }` )
+			, 50 );
+		}
+
+		if ( keyCode === BACKSPACE && canDeleteKeyword ) {
+			deleteKeyword( nonEmptyValuesLength - 1 );
+			setTimeout( () =>
+				focusInput( `#${ slug }-keyword-${ nonEmptyValuesLength - 1 }` )
+			, 50 );
+		}
+
+		if ( target.value.length <= 1 && keyCode === BACKSPACE ) {
+			// user deleted the only character, or just landed in this input after
+			// deleting a keyword so hitting backspace again should delete the previous keyword
+			setCanDeleteKeyword( true );
+		} else {
+			// user is typing, so backspace should delete the last character
+			setCanDeleteKeyword( false );
 		}
-	}, [ keywordsContainer.current, ...dependencies ] );
+	}, [ keywordsContainer.current, ...dependencies, canDeleteKeyword ] );
+
+	const focusInput = ( querySelector ) => {
+		const input = keywordsContainer.current.querySelector( querySelector );
+		if ( input ) {
+			input.focus();
+		}
+	};
```

Comment:
It's a bit confusing that this is defined after it is referenced above. I know it's technically okay and such—it isn't used until later.

But if we can move this up it might make it a bit easier to mentally parse is all. 🤷🏻‍♂️

Reply:
I agree - I've moved all these declarations higher up.